source/mir/las/sum.d

```diff
@@ -1822,16 +1828,20 @@ nothrow unittest
         version(X86)
         {
             assert(summator.isNaN() == r.isNaN());
-            assert(summator.isFinite() == r.isFinite() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(summator.isInfinity() == r.isInfinity() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(nextDown(s) <= r && r <= nextUp(s) || s.isNaN && r.isNaN);
+            assert(summator.isFinite() == r.isFinite()
+                 || (r == -double.max && s == -double.infinity)
+                 || (r == double.max && s == double.infinity));
+            assert(summator.isInfinity() == r.isInfinity()
+                 || (r == -double.max && s == -double.infinity)
+                 || (r == double.max && s == double.infinity));
+            assert((nextDown(s) <= r && r <= nextUp(s)) || (s.isNaN && r.isNaN));
```

Comment:
dscanner checks for such tricky cases where the prevalence of `&&` over `||` matters.